### PR TITLE
Android 7.2.0

### PR DESCRIPTION
**Pull request by cthomas-newrelic**
# New In the release
* Adds instrumentation for Apollo GraphQL server responses
* Adds support for user-defined header instrumentation
* Adds package-level instrumentation exclusion

## Fixed in this release
* Fixes build exception in Gradle plugin when instrumenting SHA-signed artifacts
* Excludes SHA signed artifacts from instrumentation
* 
### Support Statement
As of this release, the oldest supported version is [6.4.0](https://docs.newrelic.com/docs/release-notes/mobile-release-notes/android-release-notes/android-640/).

### CODE REVIEW

File: .github/workflows/sonatypeDeployment.yml [hunk 1]
```diff
@@ -46,3 +46,25 @@ jobs:
             ORG_GRADLE_PROJECT_signingKey: ${{ secrets.SIGNING_KEY }}
           run: ./gradlew clean publishToSonatype closeSonatypeStagingRepository --max-workers 1
 
+  publish_mono_release:
```
Comment (cthomas-newrelic) [inline .github/workflows/sonatypeDeployment.yml:49]: Will likely have to integrate static artifact deploy into normal agent publishing job

File: agent-core/src/main/java/com/newrelic/agent/android/instrumentation/TransactionState.java [hunk 2]
```diff
@@ -229,7 +242,7 @@ TransactionData toTransactionData() {
                     bytesSent,
                     bytesReceived,
                     appData,
-                    wanType, trace, null);
+                    wanType, trace, "", params, null);
```
Comment (cthomas-newrelic) [inline agent-core/src/main/java/com/newrelic/agent/android/instrumentation/TransactionState.java:245]: @ndesai-newrelic To check and refactor

File: agent/src/main/java/com/newrelic/agent/android/NewRelic.java [hunk 3]
```diff
@@ -1059,4 +1060,9 @@ public static String currentSessionId() {
     public static boolean recordJSErrorException(StackTrace stackTrace) {
         return DataController.sendAgentData(stackTrace);
     }
+
+
+    public static boolean  addHTTPHeadersTrackingFor(List<String> headers) {
```
Comment (cthomas-newrelic) [inline agent/src/main/java/com/newrelic/agent/android/NewRelic.java:1065]: @ndesai-newrelic Public docs needed for new API

File: plugins/gradle/src/functional/groovy/com/newrelic/agent/android/PluginJDK17SmokeSpec.groovy [hunk 4]
```diff
@@ -42,7 +42,7 @@ class PluginJDK17SmokeSpec extends PluginSpec {
                         "-Pcompiler=r8",
                         "-PagentRepo=${localEnv["M2_REPO"]}",
                         "-PwithProductFlavors=true",
-                        "--debug",
+                        // "--debug",
```
Comment (cthomas-newrelic) [inline plugins/gradle/src/functional/groovy/com/newrelic/agent/android/PluginJDK17SmokeSpec.groovy:45]: @cthomas - should not be here

**Review by ywang-nr — APPROVED** [review]
Nisarg and I have validated the GraphQL implementation, we are good to go.

**Review by ndesai-newrelic — APPROVED** [review]

**Pull request merged**